packages/components/src/tools-panel/tools-panel/hook.ts

```diff
@@ -110,6 +109,40 @@ export function useToolsPanel(
 		} );
 	};
 
+	// Track whether all optional controls are displayed or not.
+	// If no optional controls are present, then none are hidden and this will
+	// be `false`.
+	const [
+		areAllOptionalControlsHidden,
+		setAreAllOptionalControlsHidden,
+	] = useState( false );
+
+	// Where no optional menu items are active, we display a plus icon
+	// to indicate the presence of further menu items.
+	useEffect( () => {
+		if ( menuItems.optional ) {
+			const optionalItems = Object.entries( menuItems.optional );
+			const allControlsHidden =
+				optionalItems.length > 0 &&
+				! optionalItems.some( ( [ , isSelected ] ) => isSelected );
+			setAreAllOptionalControlsHidden( allControlsHidden );
+		}
+	}, [ menuItems.optional ] );
+
+	const hasDefaultMenuItems =
+		menuItems?.default && !! Object.keys( menuItems?.default ).length;
+
+	const wrapperStyle = hasInnerWrapper && styles.ToolsPanelWithInnerWrapper;
+	const emptyStyle =
+		! hasDefaultMenuItems &&
+		areAllOptionalControlsHidden &&
+		styles.ToolsPanelHiddenInnerWrapper;
+
+	const cx = useCx();
+	const classes = useMemo( () => {
+		return cx( styles.ToolsPanel, wrapperStyle, emptyStyle, className );
+	}, [ className, emptyStyle, wrapperStyle ] );
```

Comment:
Let's move all of the classname-related computation into the `useMemo` hook

```suggestion
	const cx = useCx();
	const classes = useMemo( () => {
		const hasDefaultMenuItems =
			menuItems?.default && !! Object.keys( menuItems?.default ).length;

		const wrapperStyle =
			hasInnerWrapper && styles.ToolsPanelWithInnerWrapper;
		const emptyStyle =
			! hasDefaultMenuItems &&
			areAllOptionalControlsHidden &&
			styles.ToolsPanelHiddenInnerWrapper;

		return cx( styles.ToolsPanel, wrapperStyle, emptyStyle, className );
	}, [
		className,
		hasInnerWrapper,
		menuItems,
		areAllOptionalControlsHidden,
	] );
```